packages/components/src/templates/next/components/complex/InfoCards/InfoCards.tsx

```diff
@@ -1,51 +1,162 @@
+  imageAlt,
+}: Pick<SingleCardWithImageProps, "imageUrl" | "imageAlt">): JSX.Element => (
+  <div className={compoundStyles.cardImageContainer()}>
+    <img src={imageUrl} alt={imageAlt} className={compoundStyles.cardImage()} />
+  </div>
+)
+
+const InfoCardText = ({
+  title,
+  description,
+  url,
+}: Pick<
+  SingleCardWithImageProps,
+  "title" | "description" | "url"
+>): JSX.Element => (
+  <div className={compoundStyles.cardTextContainer()}>
+    <h4 className={compoundStyles.cardTitle({ isClickableCard: !!url })}>
+      {title}
+
+      {url && (
+        <BiRightArrowAlt
+          aria-hidden
+          className={compoundStyles.cardTitleArrow()}
+        />
       )}
-    </div>
+    </h4>
+    <p className={compoundStyles.cardDescription()}>{description}</p>
+  </div>
+)
+
+const InfoCardNoImage = ({
+  title,
+  description,
+  url,
+  LinkComponent,
+}: SingleCardNoImageProps): JSX.Element => {
+  return (
+    <InfoCardContainer url={url} LinkComponent={LinkComponent}>
+      <InfoCardText title={title} description={description} url={url} />
+    </InfoCardContainer>
   )
 }
 
-const InfoCards = ({ cards, title, subtitle }: InfoCardsProps) => {
+const InfoCardWithImage = ({
+  title,
+  description,
+  imageUrl,
+  imageAlt,
+  url,
+  LinkComponent,
+}: SingleCardWithImageProps): JSX.Element => {
   return (
-    <section>
-      <div
-        className={`${ComponentContent} mx-auto flex flex-col items-center py-12 lg:py-24`}
-      >
-        <TitleSection title={title} subtitle={subtitle} />
-        <div className={`grid grid-cols-1 gap-8 md:grid-cols-2 lg:grid-cols-3`}>
-          {cards.map((card) => (
-            <Card
-              title={card.title}
-              url={card.url}
-              imageUrl={card.imageUrl}
-              description={card.description}
-              imageAlt={card.imageAlt}
-              buttonLabel={card.buttonLabel}
-              variant="vertical"
-            ></Card>
-          ))}
-        </div>
-      </div>
-    </section>
+    <InfoCardContainer url={url} LinkComponent={LinkComponent}>
+      <InfoCardImage imageUrl={imageUrl} imageAlt={imageAlt} />
+      <InfoCardText title={title} description={description} url={url} />
+    </InfoCardContainer>
   )
 }
 
+const InfoCards = ({
+  title,
+  subtitle,
+  variant,
+  cards,
+  LinkComponent,
+}: InfoCardsProps): JSX.Element => (
+  <section className={compoundStyles.container()}>
+    {(title || subtitle) && (
+      <InfoCardsHeadingSection title={title} subtitle={subtitle} />
+    )}
+
+    <div className={compoundStyles.grid()}>
+      {variant === "cardsWithImages" &&
+        cards.map((card, idx) => (
+          <InfoCardWithImage
+            key={idx}
+            {...card}
+            LinkComponent={LinkComponent}
+          />
+        ))}
+
+      {variant === "cardsWithoutImages" &&
+        cards.map((card, idx) => (
+          <InfoCardNoImage key={idx} {...card} LinkComponent={LinkComponent} />
+        ))}
```

Comment:
nit: should we strengthen the typechecking so build-time errors will surface if we add more variants?

```tsx
// The typing forces compile time rendering to prevent `undefined` from being returned,
// which can happen if variants change or is added
const InfoCardToRender: ReactElement = useMemo(() => {
  switch (variant) { // I assuming variant is typed like `"cardsWithImages" | "cardsWithoutImages"`
    case "cardsWithImages"
      return InfoCardWithImage
    case "cardsWithoutImage"
      return InfoCardNoImage
  }
}, [variant])

// Usage
{cards.map((card, idx) => (
  <InfoCardToRender ... />
)}
```

Reply:
Updated in 1d3845efe6d4a8db0faf455d1c260a04b0d37ea1, can't really switch case after the map cos the type doesn't narrow properly.